analyses/tp53_nf1_score/00-tp53-nf1-alterations.R

```diff
@@ -0,0 +1,103 @@
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+source(file.path(root_dir, "analyses", "snv-callers", "util",
+                 "tmb_functions.R"))
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-s", "--snvConsensus"),type="character",
+              help="Consensus snv calls (.tsv) "),
+  make_option(c("-c","--clinicalFile"),type="character",
+              help="clinical file for all samples (.tsv)"),
+  make_option(c("-o","--outputFolder"),type="character",
+              help="output folder for results "),
+  make_option(c("-g","--gencode"),type="character",
+              help="cds gencode bed file")
+)
+
+opt <- parse_args(OptionParser(option_list=option_list))
+snvConsensusFile <- opt$snvConsensus
+clinicalFile <- opt$clinicalFile
+outputFolder <- opt$outputFolder
+gencodeBed <- opt$gencode
+
+#### Generate files with TP53, NF1 mutations -----------------------------------
+
+# read in consensus SNV files
+consensus_snv <- read_tsv(snvConsensusFile)
+# gencode cds region BED file
+gencode_cds <- read_tsv(gencodeBed, col_names = FALSE)
+# clinical file
+clinical <- read_tsv(clinicalFile)
+
+# filter the MAF data.frame to only include entries that fall within the
+# CDS bed file regions
+coding_consensus_snv <- snv_ranges_filter(maf_df = consensus_snv,
+                                          keep_ranges = gencode_cds)
+
+# subset to TP53, removing silent mutations and mutations in introns
+tp53_coding <- coding_consensus_snv %>%
+  filter(Hugo_Symbol == "TP53") %>%
+  filter(!(Variant_Classification %in% c("Silent", "Intron")))
+
+# subset to NF1, removing silent mutations, mutations in introns, and missense
+# mutations -- we exclude missense mutations because they are not annotated
+# with OncoKB
+# https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/381#issuecomment-570748578
+nf1_coding <- coding_consensus_snv %>%
+  filter(Hugo_Symbol == "NF1") %>%
+  filter(!(Variant_Classification %in% c("Silent",
+                                         "Intron",
+                                         "Missense_Mutation")))
+
+# include only the relevant columns from the MAF file
+tp53_nf1_coding <- tp53_coding %>%
+  bind_rows(nf1_coding) %>%
+  select(Chromosome, Start_Position, End_Position, Strand,
+         Variant_Classification, Tumor_Sample_Barcode, Hugo_Symbol)
+
+# biospecimen IDs for tumor or cell line DNA-seq
+bs_ids <- clinical %>%
+  filter(sample_type != "Normal",
+         experimental_strategy != "RNA-Seq") %>%
+  pull(Kids_First_Biospecimen_ID)
+
+# all BS ids that are not in the data frame that contain the TP53 and NF1
+# coding mutations should be labeled as not having either
+bs_ids_without_mut <- setdiff(bs_ids,
+                              unique(tp53_nf1_coding$Tumor_Sample_Barcode))
+
+# create a data.frame with wildtype BS IDs for joining
+without_mut_df <- data.frame(
+  Chromosome = NA,
```

Comment:
I *believe* `bind_rows` [default is to fill in NAs for any columns that don't have matches](https://www.rdocumentation.org/packages/dplyr/versions/0.5.0/topics/bind) so I *believe* you can get rid of lines 89 - 93. But definitely test this before using it.